hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java

```diff
@@ -192,20 +192,17 @@ private EntityManagerFactoryBuilderImpl(
 
 		this.persistenceUnit = persistenceUnit;
 
-		if ( integrationSettings == null ) {
-			integrationSettings = Collections.emptyMap();
-		}
+		final Map integtrationSettingsCopy = integrationSettings == null ? new HashMap() : new HashMap( integrationSettings );
 
-		Map mergedIntegrationSettings = null;
 		Properties properties = persistenceUnit.getProperties();
 		if ( properties != null ) {
-			mergedIntegrationSettings = new HashMap( persistenceUnit.getProperties() );
-			mergedIntegrationSettings.putAll( integrationSettings );
+			// original integration settings take precedence
+			properties.forEach( integtrationSettingsCopy::putIfAbsent );
```

Comment:
I do think the above code makes the subtle logic explicit ending up with better code readability and maintainability. Maybe it is worthwhile to log something is some property is ignored (which might be pretty troublesome to investigate why something is not working as expected).